Add a method to get all the CRL entries as a HashMap.

weinbergerd opened the pull request:
This is useful when verifying a lot of certificates, since the CRL rarely changes and we avoid unnecessary HTTP requests which are slow.

Tested manually - no unit tests since that method requires network access.

eranmes commented:
Per OOB discussion, let's have two variants of this method, one that takes an InputStream and one that doesn't take anything (and fetches the CRL via https).

That way we could test that converting to a map does not drop any entries, etc.

weinbergerd commented:
Updated the PR accordingly (used a Reader as in the other methods of the class instead of an InputStream though), and added a test. Please look over it again.
